pkg/storaged/details.js

```diff
@@ -48,7 +48,24 @@
                 return client.mdraids[path].Start({ "start-degraded": { t: 'b', v: true } });
             },
             mdraid_stop: function mdraid_stop(path) {
-                return client.mdraids[path].Stop({});
+                var mdraid = client.mdraids[path];
+                var block = client.mdraids_block[path];
+                var usage = block? utils.get_usage_alerts(client, block.path) : [ ];
+
+                if (usage.length === 0)
+                    return mdraid.Stop({});
+
+                dialog.open({ Title: cockpit.format(_("Please confirm stopping of $0"),
+                                                    utils.mdraid_name(mdraid)),
```

Comment:
Agreed to the "we don't know everything" approach. But as this is not an "are you sure?" confirmation (as the  array is just stopped without confirmation if there isn't any usage that we can detect), it seems to me that it might be a better user experience to present `utils.get_usage_alerts()` only after stopping actually fails, and then just make it an error dialog without actions.